processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/PartitionContexts.java

```diff
@@ -139,11 +139,13 @@ public void destroyAllProcessors() {
     @Override
     public Map<TopicPartition, OffsetAndMetadata> commitReadyOffsets() {
         Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>();
-        for (PartitionContext context : contexts.values()) {
-            context.offsetWaitingCommit().ifPresent(
-                    offset -> offsets.put(context.topicPartition(),
-                                          new OffsetAndMetadata(offset + 1, null)));
-        }
+        contexts.values()
+                .stream()
+                .filter(context -> !context.revoking()).forEach(context -> {
```

Comment:
(just to confirm) there's no issue with threading, right? same thread will update this `revoking ` flag and call `commitReadyOffsets`

Reply:
Yeah, all interactions are only by subscription thread so no problem.